src/commit-subject-parser.ts

```diff
@@ -129,10 +127,10 @@ export class CommitSubjectParser {
         .trim()
     )
 
-    if (isNaN(jobId)) {
+    if (isNaN(jobNumber)) {
       throw new MissingJobIdInCommitSubjectError(this.text)
     }
 
-    return jobId
+    return new JobId(jobNumber)
```

Comment:
hi @yeraydavidrodriguez the `JobId` class has its own validation [here](https://github.com/Nautilus-Cyberneering/git-queue/pull/179/files/7993e51329cdce4c1b0a95a5ef0a9670ea8c1f0f..74842149bc70a0fa0f1b2b20aef6e88053338434#diff-9d176adb81160c3cdd199889d6dc26cc4a23fed10b6e68de1d29a58a9c17b084R16). I think we should relay on the class validation. We could catch the `JobId` exception (`InvalidJobIdError`) and throw the other one (MissingJobIdInCommitSubjectError) if you want. Althougth I would only throw the `MissingJobIdInCommitSubjectError` error when there is no `JobId` in the subject. 

I think we have a duplication for the `JobId` validation. If the `JobId` implementation changes later to UUID, for example, you have to change the lines:

```typescript
  if (isNaN(jobNumber)) {
    throw new MissingJobIdInCommitSubjectError(this.text)
  }
```

Reply:
Yes, you are right.